google/cloud/security/common/gcp_type/firewall_rule.py

```diff
@@ -195,40 +213,49 @@ def is_equivalent(self, other):
         """
         return (self.direction == other.direction and
                 self.network == other.network and
-                self.source_tags == other.source_tags and
-                self.target_tags == other.target_tags and
+                self._source_tags == other._source_tags and
+                self._target_tags == other._target_tags and
                 self.source_ranges == other.source_ranges and
                 self.destination_ranges == other.destination_ranges and
                 self.firewall_action.is_equivalent(other.firewall_action))
+    # pylint: enable=protected-access
 
 
 class FirewallAction(object):
     """An association of allowed or denied ports and protocols."""
 
-    def __init__(self, firewall_rule_allowed=None, firewall_rule_denied=None):
+    ALLOW = 'allow'
+    DENY = 'deny'
+    ANY_VALUE = '*'
+
+    def __init__(self, firewall_rules=None, firewall_rule_action='allow'):
         """Initialize.
 
         Args:
-          firewall_rule_allowed (list): A list of dictionaries of allowed ports
-            and protocols.
-          firewall_rule_denied (list): A list of dictionaries of denied ports
+          firewall_rules (list): A list of dictionaries of allowed ports
             and protocols.
+          firewall_rule_action (str): The action, either ALLOW or DENY.
 
         Raises:
           ValueError: If there are both allow and deny rules.
         """
-        if firewall_rule_allowed:
-            if firewall_rule_denied:
-                raise ValueError(
-                    'Rule cannot have deny (%s) and allow (%s) actions' %
-                    (firewall_rule_denied, firewall_rule_allowed))
-            self.action = 'allow'
-            self.rules = sort_rules(firewall_rule_allowed)
+        if firewall_rule_action not in (FirewallAction.ALLOW,
+                                        FirewallAction.DENY):
+            raise ValueError(
+                'Firewall rule action must be either ALLOW or DENY, got: %s' % (
```

Comment:
Shouldn't it be 'allow' or 'deny' ? Capitals aren't allowed.

Reply:
Done